### PR TITLE
Added more customization points for subclasses

### PR DESCRIPTION
**Pull request by kobsy**
For the CollectionEditor, adds `alternateRows`. For the ModelEditor, adds `waitOnSave` and the method `processAttributes`.

### CODE REVIEW

File: vendor/assets/javascripts/neat/collection_editor.coffee [hunk 1]
```diff
@@ -142,7 +143,7 @@ class window.Neat.CollectionEditor extends Backbone.View
       @views.push(view)
 
       $el = $(view.render().el)
-      $el.toggleClass 'alt', !(alt = !alt)
+      $el.toggleClass 'alt', !(alt = !alt) if @alternateRows
```
Comment (boblail) [inline vendor/assets/javascripts/neat/collection_editor.coffee:146]: What if it always added `alt` but you just didn't use CSS to style it?

Or does it conflict with something else?

File: vendor/assets/javascripts/neat/model_editor.coffee [hunk 2]
```diff
@@ -82,7 +84,7 @@ class window.Neat.ModelEditor extends Backbone.View
       previousAttributes = @model.toJSON()
 
       @model.save attributes,
-        wait: true
+        wait: @waitOnSave
```
Comment (boblail) [inline vendor/assets/javascripts/neat/model_editor.coffee:86]: :+1: So good

File: vendor/assets/javascripts/neat/model_editor.coffee [hunk 3]
```diff
@@ -73,6 +74,7 @@ class window.Neat.ModelEditor extends Backbone.View
   save: (e)->
     e?.preventDefault()
     newAttributes = @attributesFromForm(@$el)
+    newAttributes = @processAttributes(newAttributes)
```
Comment (boblail) [inline vendor/assets/javascripts/neat/model_editor.coffee]: What if you just overrode `attributesFromForm` to be something like this?

``` coffee
attributesFromForm: ($el) ->
  attributes = super $el
  # my custom processing code
  attributes
```

File: vendor/assets/javascripts/neat/collection_editor.coffee (hunk 1 above)
Comment (kobsy) [inline vendor/assets/javascripts/neat/collection_editor.coffee:146]: This was actually causing issues with styling some of the elements contained in an `alt` line. For example, if you had `.alt > .bubble`. CSS could be used to un-style these descendent elements, but there would be a lot less to counteract if the class wasn't added in the first place.

File: vendor/assets/javascripts/neat/model_editor.coffee (hunk 3 above)
Comment (kobsy) [inline vendor/assets/javascripts/neat/model_editor.coffee]: You're quite right. I'll remove this change.

**Comment by boblail** [comment]
Merged (on the command line)

**Pull request closed without merging**